lib/Parse/ParseDecl.cpp

```diff
@@ -1460,6 +1347,41 @@ bool Parser::parseDeclAttribute(DeclAttributes &Attributes, SourceLoc AtLoc) {
         .fixItReplace(Tok.getLoc(), "available");
   }
 
+
+  if (DK == DAK_Count && Tok.getText() == "warn_unused_result") {
```

Comment:
Would you be okay with my adding that new "removed attribute" case in this PR? I'd rather do that than re-introduce those cases everywhere. Alternatively, if you would like to add the attribute, I'd be happy to use it.

If my proposal to remove `@noescape` is accepted, there will be at least one other attribute in a very similar situation.